tests/test_mig_shared_functionality_cat.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+#
+# --- BEGIN_HEADER ---
+#
+# test_mig_shared_functionality_cat - cat functionality unit test
+# Copyright (C) 2003-2024  The MiG Project by the Science HPC Center at UCPH
+#
+# This file is part of MiG.
+#
+# MiG is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# MiG is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+# USA.
+#
+# --- END_HEADER ---
+#
+
+"""Unit tests of the MiG functionality file implementing the cat resource."""
+
+from __future__ import print_function
+import importlib
+import os
+import shutil
+import sys
+
+from tests.support import MIG_BASE, MigTestCase, testmain, \
+    fixturefile, fixturefile_normname, \
+    _ensuredirs, _temppath
```

Comment:
Minor: private and semi-private functions like these (one or two leading underscores) should generally _not_ be used outside the module itself. 
While it's not a blocker they should really either be made fully public (i.e. strip leading underscore and make sure they are generally usable) or the import replaced by existing public functions. The `shared.fileio` already has the `makedirs_rec` function, which appears to overlap some 90%  percent with `_ensuredirs`.
I'll take a look and either adjust during merge or add the follow-up marker here.